Doc/library/os.rst

```diff
@@ -1659,6 +1659,33 @@ or `the MSDN <https://msdn.microsoft.com/en-us/library/z0kc8e3z.aspx>`_ on Windo
       :exc:`InterruptedError` exception (see :pep:`475` for the rationale).
 
 
+.. function:: readinto(fd, buffer, /)
+
+   Read from a file descriptor *fd* into a mutable
+   :ref:`buffer protocol <bufferobjects>` *buffer*.
+
+   The *buffer* should be mutable and :term:`bytes-like <bytes-like object>`. On
+   success, returns the number of bytes read. Less bytes may be read than the
+   size of the buffer. The underlying system call will be retried when
+   interrupted by a signal. Other errors will not be retried and an error will
+   be raised.
+
+   Returns 0 if the fd is at end of file or if the provided buffer is
+   length 0 (can be used to check for errors without reading data). Never
+   returns a negative value.
+
+   .. note::
+
+      This function is intended for low-level I/O and must be applied to a file
+      descriptor as returned by :func:`os.open` or :func:`os.pipe`.  To read a
+      "file object" returned by the built-in function :func:`open`, or
+      :data:`sys.stdin`, use its member functions, for example
+      :meth:`io.BufferedIOBase.readinto`, :meth:`io.BufferedIOBase.read`, or
+      :meth:`io.TextIOBase.read`
```

Comment:
This note is a little bit surprising. We don't have it for other os functions. Is it really useful?

Reply:
I got it from [`os.read`](https://docs.python.org/3/library/os.html#os.read) guessing it was added there because people kept asking. Others like `os.pread` don't have analogs on io objects which I think make it a less common switch for them. readinto exists on _some_ io objects but not all (ex. `BufferedIO` and `FileIO` have `readinto` but TextIO does not)

Reply:
Oh, I didn't notice os.read() note. In this case, you can keep it :-)